### PR TITLE
[issue#671] Fix smart contract code generators by generating two more…

### PR DESCRIPTION
**Pull request by franz-see**
… constructors on the SolidityFunctionWrapper generated classes:

 * protected <Contract>(String contractAddress, Web3j web3j, Credentials credentials, ContractGasProvider gasProvider) {}
 * protected <Contract>(String contractAddress, Web3j web3j, TransactionManager transactionManager, ContractGasProvider gasProvider) {}

### CODE REVIEW

File: core/src/main/java/org/web3j/tx/Contract.java [hunk 1]
```diff
@@ -86,6 +86,14 @@ protected Contract(String contractBinary, String contractAddress,
                 gasPrice, gasLimit);
     }
```
[the remark this answers is not in the document]
Comment (franz-see) [reply core/src/main/java/org/web3j/tx/Contract.java]: Fixed in 137c66a

**Comment by yuriymyronovych** [comment]
@franz-see looks great!

**Comment by iikirilov** [comment]
This has been addressed in #725

**Comment by stewartsvg** [comment]
Hi, the new 4.0.1 web3j CLI tool smart contract wrappers generate a ContractGasProvider, but the latest Android version (3.3.1) of web3j do not contain this

**Comment by stewartsvg** [comment]
using web3j CLI version 3.5.0 works, is that the only workaround I can use for now? Sorry if this is not the right place to ask this question

**Comment by iikirilov** [comment]
@stewartsvg please use web3j 3.5.0 or higher

we are in progress to release the android branch - I believe a snapshot is available

**Pull request closed without merging**